### PR TITLE
Fixes #25810 - migrate react-bootstrap to patternfly-react

### PR DESCRIPTION
**Pull request by MariaAga**
Refs #22001 #22232 #22006

### CODE REVIEW

**Comment by ohadlevy** [comment]
[ok to test]

**Review by amirfefer — CHANGES_REQUESTED** [review]
Thanks @MariaAga ! 
The breadcrumbs bar doesn't render at all with this change, having this js error:
```
Element type is invalid: expected a string (for built-in components) or a class/function (for composite 
components) but got: undefined. You likely forgot to export your component from the file it's defined in, or
 you might have mixed up default and named imports.

Check the render method of `BreadcrumbSwitcher`.
```

Plus, the statistics page looks different now, from this:
statistics-1
To this:
statistics-2

I like the new look though
but for next time, just upload some screenshots  if there's any change :)

**Comment by MariaAga** [comment]
@amirfefer Did rebase, and couldn't reproduce this issue.

**Comment by ohadlevy** [comment]
can we remove react-bootstrap from package.json now?

**Review by ohadlevy — DISMISSED** [review]
when trying out the PR, I've noticed the following error in my browse logs:
```
react-dom.development.js:506 Warning: validateDOMNesting(...): <h3> cannot appear as a child of <h2>.
    in h3 (created by ChartBox)
    in h2 (created by CardTitle)
    in CardTitle (created by ChartBox)
    in div (created by CardHeading)
    in CardHeading (created by ChartBox)
    in div (created by Card)
    in Card (created by ChartBox)
    in ChartBox (created by StatisticsChartsList)
    in div (created by StatisticsChartsList)
    in StatisticsChartsList (created by Connect(StatisticsChartsList))
    in Connect(StatisticsChartsList) (created by I18nProviderWrapper(Connect(StatisticsChartsList)))
    in IntlProvider (created by I18nProviderWrapper(Connect(StatisticsChartsList)))
    in I18nProviderWrapper(Connect(StatisticsChartsList)) (created by StoreProvider(I18nProviderWrapper(Connect(StatisticsChartsList))))
    in Provider (created by StoreProvider(I18nProviderWrapper(Connect(StatisticsChartsList))))
    in StoreProvider(I18nProviderWrapper(Connect(StatisticsChartsList))) (created by DataProvider(StoreProvider(I18nProviderWrapper(Connect(StatisticsChartsList)))))
    in DataProvider(StoreProvider(I18nProviderWrapper(Connect(StatisticsChartsList))))
```

**Comment by MariaAga** [comment]
@ohadlevy Removed the `Card.Title` and now its ok.
Design changed to:
screenshot-localhost-5000-2019 01 21-16-31-50

**Comment by ohadlevy** [comment]
this looks wrong...? why not use the card title?

**Comment by MariaAga** [comment]
Using the `Card.Title` makes that design,:
screenshot-localhost-5000-2019 01 22-11-46-27

Is it better? @ohadlevy

**Comment by ohadlevy** [comment]
@MariaAga can you rebase please?

File: app/models/report.rb [hunk 1]
```diff
@@ -89,22 +89,28 @@ def self.expire(conditions, batch_size, sleep_time)
         batch_start_time = Process.clock_gettime(Process::CLOCK_MONOTONIC)
         report_ids = where(cond).reorder('').limit(batch_size).pluck(:id)
         if report_ids.count > 0
-          message_count = Message.unscoped.joins(:logs).where("logs.report_id" => report_ids).delete_all
-          source_count = Source.unscoped.joins(:logs).where("logs.report_id" => report_ids).delete_all
```
Comment (ohadlevy) [inline app/models/report.rb]: why is this part of the PR?

Comment (MariaAga) [reply app/models/report.rb]: Removed. thanks

**Comment by waldenraines** [comment]
[test foreman]
[test katello]

**Review by waldenraines — DISMISSED** [review]
LGTM, @amirfefer @ohadlevy any other comments?

**Comment by mmoll** [comment]
@MariaAga please rebase

**Comment by MariaAga** [comment]
@mmoll, rebased thanks!

**Comment by mmoll** [comment]
@amirfefer @ohadlevy anything missing here?

**Comment by mmoll** [comment]
@MariaAga actually, I think the foreman test failure is related :broken_heart:

**Comment by MariaAga** [comment]
@mmoll Thank you! I've change the statistics page test as it was looking for h3 titles instead of the new h2 titles

**Review by Ron-Lavi — DISMISSED** [review]
Thanks @MariaAga, Looks great :+1: :trophy:

**Comment by ohadlevy** [comment]
should we remove the entry from package.json then? @sharvit ?

**Comment by ohadlevy** [comment]
bump on the package.json question please?

**Comment by mmoll** [comment]
@MariaAga also, please rebase :)

**Comment by ohadlevy** [comment]
also, I think it makes sense to add the cards-pf css class to the container to fit better https://www.patternfly.org/pattern-library/content-views/card-view

**Comment by sharvit** [comment]
Thanks @MariaAga :+1: 

> should we remove the entry from package.json then? @sharvit ?

afaik we can remove it from `package.json` and `webpack.vendor.js`

**Comment by MariaAga** [comment]
@ohadlevy, I don't think we should add this class as it causes this grey background:
image
or this (when applying in the container of the container):
image

**Comment by ohadlevy** [comment]
@MariaAga - the grey background does look better than white on white, I do wonder whats the difference in our styles that make it look different to patternfly? IMHO we should have grey backgrounds with cards, for example in a similar screen (in this case openshift)
image

**Comment by MariaAga** [comment]
@ohadlevy  How about this? (or maybe this with more padding in the container)
image

**Comment by ohadlevy** [comment]
> @ohadlevy How about this? (or maybe this with more padding in the container)

mind providing full page screenshot? regarding if its better, I would delegate to @theforeman/ui-ux designers?

**Comment by MariaAga** [comment]
Sure!
image
image

**Comment by ohadlevy** [comment]
@MariaAga please remove the bootstrap-react npm from package.json ?

**Comment by MariaAga** [comment]
Removed from package.json and webpack.vendor.js

**Review by sharvit — DISMISSED** [review]
Thanks @MariaAga LGTM :+1:

**Comment by sharvit** [comment]
@ekohl does it need a packaging PR?
`react-bootsrap` removed from core package.json but getting installed via `patternfly-react`?

**Comment by mmoll** [comment]
@ekohl @ehelms @evgeni :point_up:

**Review by mmoll — APPROVED** [review]
RPMs should be fine, AFAICT.

**Comment by mmoll** [comment]
merged, thanks @MariaAga!

**Pull request merged**